lib/ipam/ipam.go

```diff
@@ -296,21 +228,21 @@ func (c ipams) AssignIP(args AssignIPArgs) error {
 	log.Infof("Assigning IP %s to host: %s", args.IP, hostname)
 
 	if !c.blockReaderWriter.withinConfiguredPools(args.IP) {
-		return goerrors.New("The provided IP address is not in a configured pool\n")
+		return errors.New("The provided IP address is not in a configured pool\n")
 	}
 
 	blockCIDR := getBlockCIDRForAddress(args.IP)
 	log.Debugf("IP %s is in block '%s'", args.IP.String(), blockCIDR.String())
 	for i := 0; i < ipamEtcdRetries; i++ {
-		obj, err := c.client.Backend.Get(model.BlockKey{blockCIDR})
+		obj, err := c.client.Get(context.Background(), model.BlockKey{blockCIDR}, "")
```

Comment:
should we expose the context to the client? so when a client calls `AssignIP` it can pass a context and cancel it if needed? WDYT?

Reply:
We could - not sure anyone will use it though.